x-pack/filebeat/tests/integration/input_reload_test.go

```diff
@@ -0,0 +1,354 @@
+	}, waitDeadlineOr5Min(), 100*time.Millisecond,
+		"String 'Can only start an input when all related states are finished' not found on Filebeat logs")
+
+	require.Eventually(t, func() bool {
+		return filebeat.LogContains("file 'flog.log' is not finished, will retry starting the input soon")
+	}, waitDeadlineOr5Min(), 100*time.Millisecond,
+		"String 'file 'flog.log' is not finished, will retry starting the input soon' not found on Filebeat logs")
+
+	require.Eventually(t, func() bool {
+		return filebeat.LogContains("ForceReload set to TRUE")
+	}, waitDeadlineOr5Min(), 100*time.Millisecond,
+		"String 'ForceReload set to TRUE' not found on Filebeat logs")
+
+	require.Eventually(t, func() bool {
+		return filebeat.LogContains("Reloading Beats inputs because forceReload is true")
+	}, waitDeadlineOr5Min(), 100*time.Millisecond,
+		"String 'Reloading Beats inputs because forceReload is true' not found on Filebeat logs")
+
+	require.Eventually(t, func() bool {
+		return filebeat.LogContains("ForceReload set to FALSE")
+	}, waitDeadlineOr5Min(), 100*time.Millisecond,
+		"String 'ForceReload set to FALSE' not found on Filebeat logs")
+
+	// Set it to true, so the temporary directory is removed
+	testSucceeded = true
+}
+
+func requireNewStruct(t *testing.T, v map[string]interface{}) *structpb.Struct {
+	str, err := structpb.NewStruct(v)
+	if err != nil {
+		require.NoError(t, err)
+	}
+	return str
+}
+
+// generateLogFile generates a log file by appending the current
+// time to it every second.
+func generateLogFile(t *testing.T, fullPath string) {
+	t.Helper()
+	f, err := os.Create(fullPath)
+	if err != nil {
+		t.Fatalf("could not create file '%s: %s", fullPath, err)
+	}
+
+	go func() {
+		t.Helper()
+		ticker := time.NewTicker(time.Second)
+		t.Cleanup(ticker.Stop)
+
+		done := make(chan struct{})
+		t.Cleanup(func() { close(done) })
+
+		defer func() {
+			if err := f.Close(); err != nil {
+				t.Errorf("could not close log file '%s': %s", fullPath, err)
+			}
+		}()
+
+		for {
+			select {
+			case <-done:
+				return
+			case now := <-ticker.C:
+				_, err := fmt.Fprintln(f, now.Format(time.RFC3339))
+				if err != nil {
+					// The Go compiler does not allow me to call t.Fatalf from a non-test
+					// goroutine, so just log it instead
+					t.Errorf("could not write data to log file '%s': %s", fullPath, err)
+					return
+				}
+				// make sure log lines are synced as quickly as possible
+				if err := f.Sync(); err != nil {
+					t.Errorf("could not sync file '%s': %s", fullPath, err)
+				}
+			}
+		}
+	}()
+}
+
+func ensureESIsRunning(t *testing.T) {
+	t.Helper()
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(500*time.Second))
+	defer cancel()
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "http://localhost:9200", nil)
+	if err != nil {
+		t.Fatalf("cannot create request to ensure ES is running: %s", err)
+	}
+
+	req.SetBasicAuth("admin", "testing")
```

Comment:
These should be set by environment variables:

https://github.com/elastic/beats/blob/d36f48672653082fd5a8e27795726fb87c733be2/dev-tools/mage/integtest_docker.go#L183-L189

Reply:
Done on a98f0a9cef